SpatialGDK/Source/SpatialGDK/Public/EngineClasses/AbstractPackageMapClient.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialConstants.h"
+#include "WorkerSDK/improbable/c_worker.h"
+
+#include "Engine/PackageMapClient.h"
+#include "UObject/Object.h"
+
+#include "AbstractPackageMapClient.generated.h"
+
+UCLASS()
+class SPATIALGDK_API UAbstractPackageMapClient : public UPackageMapClient
+{
+	GENERATED_BODY()
+public:
+	virtual Worker_EntityId GetEntityIdFromObject(const UObject* Object) { return SpatialConstants::INVALID_ENTITY_ID;  };
```

Comment:
I wonder what if we create a free function 
`template<typename T>`
`Worker_EntityId GetEntityIdFromObject(T* PackageMap, const UObject* Object)` instead of introducing `UAbstractPackageMapClient` and `USpatialPackageMapClientMock`? Would it be more confusing?

Reply:
I'm not too sure how this would work or just am not understanding. Does this mean I have to still mock the PackageMap to pass into this function? If not, then I have to mock this function, but how?

Reply:
Yes, you'd still have to mock PackageMap, but it wouldn't require the inheritance. As I think of it - maybe it's better to do it the way you did, to make it usable for other tests. I'll think a bit more about it.